### DISCUSSION TITLE
3D viewer

### DISCUSSION THREAD

**Discussion by khyati18** [sceneview/sceneview#309] [Q&A]
Model-viewer sample is not working in latest release.

**Comment by ThomasGorisse** [comment]
For a 3D model viewer in v4.0.0-rc.1:

```kotlin
implementation("io.github.sceneview:sceneview:4.0.0-rc.1")

SceneView {
    val model = rememberModelInstance(modelLoader, "models/mymodel.glb")
    if (model != null) {
        ModelNode(modelInstance = model, scaleToUnits = 1.0f)
    }
}
```

This gives you orbit camera controls, pinch-to-zoom, and environment lighting out of the box. See `samples/android-demo` for a full working 3D viewer sample.
